pkg/collector/corechecks/gpu/gpu.go

```diff
@@ -0,0 +1,206 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+package gpu
+
+import (
+	"fmt"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/gpu/model"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	processnet "github.com/DataDog/datadog-agent/pkg/process/net"
+	sectime "github.com/DataDog/datadog-agent/pkg/security/resolvers/time"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// Check doesn't need additional fields
+type Check struct {
+	core.CheckBase
+	config         *CheckConfig
+	sysProbeUtil   *processnet.RemoteSysProbeUtil
+	lastCheckTime  time.Time
+	timeResolver   *sectime.Resolver
+	statProcessors map[uint32]*StatsProcessor
+}
+
+// Factory creates a new check factory
+func Factory() optional.Option[func() check.Check] {
+	return optional.NewOption(newCheck)
+}
+
+func newCheck() check.Check {
+	return &Check{
+		CheckBase:      core.NewCheckBase(CheckName),
+		config:         &CheckConfig{},
+		statProcessors: make(map[uint32]*StatsProcessor),
+	}
+}
+
+// Parse parses the check configuration
+func (m *CheckConfig) Parse(data []byte) error {
+	return yaml.Unmarshal(data, m)
+}
+
+// Cancel cancels the check
+func (m *Check) Cancel() {
+	ret := nvml.Shutdown()
+	if ret != nvml.SUCCESS {
```

Comment:
should we report also `[NVML_ERROR_UNINITIALIZED](https://docs.nvidia.com/deploy/nvml-api/group__nvmlDeviceEnums.html#group__nvmlDeviceEnums_1gg06fa9b5de08c6cc716fbf565e93dd3d0e054644f6612bf2ffb3f0e68379ec612)` ?

Reply:
You're right, we can ignore that error.